File: TCPServer.java
import java.io.*;
import java.net.*;

public class TCPServer {
    public static void main(String[] args) {
        int port = 9999; 
        try {
            ServerSocket serverSocket = new ServerSocket(port);
            System.out.println("Voting Server Started on port " + port);
            int candidate1Votes = 0;
            int candidate2Votes = 0;

            while (true) {
                Socket clientSocket = serverSocket.accept();
                System.out.println("New client connected: " + clientSocket.getInetAddress().getHostName());

                BufferedReader in = new BufferedReader(new InputStreamReader(clientSocket.getInputStream()));
                PrintWriter out = new PrintWriter(clientSocket.getOutputStream(), true);

                String clientInput = in.readLine();

                if (clientInput.equalsIgnoreCase("Candidate1")) {
                    candidate1Votes++;
                    out.println("Vote successfully registered for Candidate1");
                } else if (clientInput.equalsIgnoreCase("Candidate2")) {
                    candidate2Votes++;
                    out.println("Vote successfully registered for Candidate2");
                } else if (clientInput.equalsIgnoreCase("Result")) {
                    out.println("Candidate1 Votes: " + candidate1Votes + "\nCandidate2 Votes: " + candidate2Votes);
                } else {
                    out.println("Invalid input. Please try again.");
                }

                in.close();
                out.close();
                clientSocket.close();
            }

        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}

File: TCPserver.py
import socket

# Create a socket object
server_socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)

# Define the host and port
host = socket.gethostname()  # Host IP
port = 12345

# Bind the socket to the host and port
server_socket.bind((host, port))

# Listen for incoming connections
server_socket.listen(1)

print('Server listening on {}:{}'.format(host, port))

# Accept a client connection
client_socket, address = server_socket.accept()
print('Connected to client:', address)

while True:
    # Receive data from the client
    data = client_socket.recv(1024).decode()
    if not data:
        break
    
    print("Received message:", data, "from", address)

client_socket.close()
server_socket.close()
